### PR TITLE
Monitoring tasks while generating resource dump

### PR DESCRIPTION
**Pull request by vedangimittal**
- Monitoring the task while generating the resource dump to alert the user of any failures during the process.
- Defect: https://jazz07.rchland.ibm.com:13443/jazz/web/projects/CSSD#action=com.ibm.team.workitem.viewWorkItem&id=383586

### CODE REVIEW

**Comment by vedangimittal** [comment]
**Testing Steps:**

- Power on the machine and ensure it is at least in PHYP Standby mode.

- Create a resource dump and navigate to the network tab under inspect to open the task API for monitoring the 'TaskState'.

- If the 'TaskState' is 'Completed', a success toast message will be displayed.

- If the 'TaskState' is 'Running', an API at the backend will be triggered every 30 seconds (till 20 minutes) to check if the 'TaskState' changes from 'Running' to 'Completed' or 'Failed'.

- If the 'TaskState' is 'Failed', an error toast message will be displayed.

- If the 'TaskState' remains 'Running' even after 20 minutes, an error toast message will be displayed.


**For testing purposes if you need to assign the value of TaskState explicitly, use the following commands on Terminal:**

**To set the TaskState as failed:**

busctl set-property xyz.openbmc_project.Dump.Manager /xyz/openbmc_project/dump/resource/entry/_ xyz.openbmc_project.Common.Progress Status s "xyz.openbmc_project.Common.Progress.OperationStatus.Failed"

In the blank after entry/, enter the task number which is task api number+1 (For example if on api it shows task 0 in the command 1 needs to be mentioned)

**To set the TaskState as Completed:**

busctl set-property xyz.openbmc_project.Dump.Manager /xyz/openbmc_project/dump/resource/entry/_ xyz.openbmc_project.Common.Progress Status s "xyz.openbmc_project.Common.Progress.OperationStatus.Completed"

In the blank after entry/, enter the task number which is task api number+1 (For example if on api it shows task 0 in the command 1 needs to be mentioned)

**An example of changing the task state of task 0 to failed:**

busctl set-property xyz.openbmc_project.Dump.Manager /xyz/openbmc_project/dump/resource/entry/1 xyz.openbmc_project.Common.Progress Status s "xyz.openbmc_project.Common.Progress.OperationStatus.Failed"

File: src/views/Logs/Dumps/DumpsForm.vue [hunk 1]
```diff
@@ -178,6 +180,47 @@ export default {
     };
   },
   methods: {
+    async checkTask() {
+      //getting list of all tasks and getting the api to the most recent task
+      await api.get('/redfish/v1/TaskService/Tasks').then(({ data }) => {
```
Comment (Nikhil-Ashoka) [inline src/views/Logs/Dumps/DumpsForm.vue]: This API, Can we access from the Store instead of here?

**Review by Nikhil-Ashoka — CHANGES_REQUESTED** [review]

File: src/views/Logs/Dumps/DumpsForm.vue [hunk 2]
```diff
@@ -178,6 +180,47 @@ export default {
...
+          } else if (taskState == 'Cancelled') {
+            this.errorToast(this.$t('pageDumps.toast.resourceDumpFailed'));
+          }
+        });
+      };
+      //trigger funtion to check 'TaskState'
+      if (taskLink) {
+        currentTaskProgress(0);
+      } else {
+        return this.errorToast('error');
```
Comment (Nikhil-Ashoka) [inline src/views/Logs/Dumps/DumpsForm.vue]: This will just give "error" and doesn't print the actual error message

**Review by Nikhil-Ashoka — CHANGES_REQUESTED** [review]

**Review by nabilpandiyalayil — APPROVED** [review]

**Review by Nikhil-Ashoka — APPROVED** [review]
Looks Good!

**Pull request merged**